03week/mastermind.js

```diff
@@ -42,26 +42,26 @@ function generateHint(userGuess) {
       userGuessArr[i]= '';
     }
   }
- for(let i=0; i<userGuessArr.length; i++) {
-   if(userGuessArr[i]) {
-     for (let j=0; j<solutionArr.length; j++) {
-       if (userGuessArr[i] === solutionArr[j]) {
-         correctLetter++;
-         solutionArr[j] = '';
-       }
-     }
-   }
- }
+  for(let i=0; i<userGuessArr.length; i++) {
+    if(userGuessArr[i]) {
```

Comment:
No need to run `if` statement here since this will only run through `userGuessArr.length`